File: fixtures/scale/objc/large/Families/Beaufort/BeaufortDescriptor.m
#import "BeaufortDescriptor.h"
#import "BeaufortCipher.h"
#import "BeaufortVectors.h"

@implementation BeaufortDescriptor

- (NSString *)family {
    return @"beaufort";
}

- (NSString *)suite {
    return @"polyalphabetic";
}

- (id<CLCipher>)cipher {
    return [[BeaufortCipher alloc] init];
}

- (NSArray *)vectors {
    return [BeaufortVectors all];
}

@end

File: fixtures/scale/php/large/src/Families/Beaufort/BeaufortDescriptor.php
<?php

namespace CipherLab\Families\Beaufort;

use CipherLab\Abstractions\Cipher;
use CipherLab\Abstractions\FamilyDescriptor;

// Registered once in FamilyCatalog; ties together BeaufortCipher and
// BeaufortVectors under the "polyalphabetic" suite.
class BeaufortDescriptor implements FamilyDescriptor
{
    public function family(): string
    {
        return 'beaufort';
    }

    public function suite(): string
    {
        return 'polyalphabetic';
    }

    public function cipher(): Cipher
    {
        return new BeaufortCipher();
    }

    public function vectors(): array
    {
        return BeaufortVectors::all();
    }
}
